### PR TITLE
Parents page v2

### PR DESCRIPTION
**Pull request by jmif**
- Removes some old Vue routes that are for ozaria only
- Adds new parents-v2 route for new parent page design
- Adds a basic parents page with CTA hooks in place
- Adds a BackboneModalHarness Vue component to allow us to use Backbone modals in Vue

### CODE REVIEW

File: app/core/Router.coffee [hunk 1]
```diff
@@ -138,9 +138,6 @@ module.exports = class CocoRouter extends Backbone.Router
     'editor/article': go('editor/article/ArticleSearchView')
     'editor/article/preview': go('editor/article/ArticlePreviewView')
     'editor/article/:articleID': go('editor/article/ArticleEditView')
-    'editor/cinematic(/*subpath)': go('core/SingletonAppVueComponentView')
```
Comment (AndrewJakubowicz) [inline app/core/Router.coffee:141]: Excellent cleanup!

File: app/core/store/modules/products.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+  },
+
+  actions: {
+    loadProducts ({ commit }) {
+      commit('toggleLoadingProducts')
+
+      return new Products()
+        .fetch()
+        .done(products => commit('addProducts', { products }))
```
Comment (AndrewJakubowicz) [inline app/core/store/modules/products.js:34]: For my own understanding. Is the `done`, `fail`, and `always` hooks something the CocoCollections model exposes?

File: app/views/core/SingletonAppVueComponentView.js [hunk 3]
```diff
@@ -24,7 +24,12 @@ export default class SingletonAppVueComponentView extends VueComponentView {
       store,
       router: this.router,
 
-      render: (h) => h(Root)
+      render: (h) => h(Root),
+
+      provide: {
```
Comment (AndrewJakubowicz) [inline app/views/core/SingletonAppVueComponentView.js:29]: This is so cool. Wow.

File: app/views/common/BackboneViewHarness.vue [hunk 4]
```diff
@@ -3,6 +3,15 @@
 </template>
 
 <script>
+  /**
+   * Note a known limitation of this harness is that it does not support swapping out the following props
```
Comment (AndrewJakubowicz) [inline app/views/common/BackboneViewHarness.vue:7]: Could this be resolved with a watcher, multiple watchers?

File: app/views/common/BackboneModalHarness.vue [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        this.openLegacyModal(this.modalViewInstance)
+      },
+
+      closeModal() {
+        if (!this.modalViewInstance) {
+          return
+        }
+
+        this.modalViewInstance.destroy()
+        this.cleanupModal()
```
Comment (AndrewJakubowicz) [inline app/views/common/BackboneModalHarness.vue:74]: A naive question, but does `destroy` keep a reference to something that can have events removed?
I sort of expect that events would get removed first. Then `destroy` called, and finally variable made undefined.

File: app/core/store/modules/products.js (hunk 2 above)
Comment (jmif) [reply app/core/store/modules/products.js:34]: Yup, I was hoping it was a promise but unfortunately it's a jqXHR object

File: app/views/core/SingletonAppVueComponentView.js (hunk 3 above)
Comment (jmif) [reply app/views/core/SingletonAppVueComponentView.js:29]: Yes agreed!  I learned about this too when trying to find ways to expose the CocoView methods to Vue, really excited about it, super clean.

File: app/views/common/BackboneViewHarness.vue (hunk 4 above)
Comment (jmif) [reply app/views/common/BackboneViewHarness.vue:7]: Yeah definitely I initially tried to use an all props watcher but those aren't supported.  I spent some time thinking about this and realized that for the most part I don't think that we need this functionality.  These arguments really should only be run when the component (or view) is mounted, changing them means that a significant view change is happening (swapping out the full view in the harness) or the view arguments have changed.  If the view arguments have changed they'd then be passed to the constructor which would require a full view reload.  So the semantics of the arguments changing for the View and the semantics of the props changing for the Vue component don't really line up.  If the arguments to the View constructor changes, the View would also need to support "updated" methods and there is no API for this in backbone views.   Given this I decided to add as a known limitation and wait until we need something more complex.

Tried to capture all the thoughts there but not sure I covered the nuance in how Vue components load vs how View components load, lmk if not and happy to discuss IRL.

File: app/views/common/BackboneModalHarness.vue (hunk 5 above)
Comment (jmif) [reply app/views/common/BackboneModalHarness.vue:74]: Yeah it does but I like your thinking I'll rearrange.

**Comment by jmif** [comment]
@AndrewJakubowicz thanks!  no risk right now it's on its own isolated URL

**Pull request merged**